### ISSUE TITLE
Not an issue but an implementation

### ISSUE THREAD

**Issue by rneurink** [omersiar/esp-rfid#7]
Not an issue but I've been working on my own RFID system and i thought lets merge these projects. So heres some code to implement a timing system

The timing system makes it available so someone can select times for each day to allow someone to enter.
![capture](https://user-images.githubusercontent.com/21998181/28371372-19655e12-6c9d-11e7-9bc9-3ce1b2b7f32f.PNG)

I'm still working on converting my logging system to this project. Feel free to ask questions

Code is in attached zip file

[RFID_ESP.zip](https://github.com/omersiar/esp-rfid/files/1159353/RFID_ESP.zip)

Edit:
Also it might be handy to use Gitt to improve communication around your projects

**Comment by omersiar** [comment]
- [ ] Add Timed entry
- [x] Add Captive Portal 

from @rneurink 's fork

- [x] Add ability to choose administrative password
- [x] Add ability to choose time zone, NTP Server, Browser Time Sync (for AP Mode)

What else for v0.2?

**Comment by akeilox** [comment]
This project is coming along very nicely since I last checked, loving the direction being taken!

Are they any plans to change the SS and RST pin selection (esp if rst is redundant now) in GUI to Enter and Exit pin selection so it can support 2 mfrc522 units connected to the same mcu? Perhaps the default is Enter and Exit being optional.
So one can do logs like;

Name BOB Card A1235B Entered [Door-Name] at [Date and Time]
...
Name BOB Card A1235B Exited [Door-Name] at [Date and Time]

**Comment by omersiar** [comment]
Hello @akeilox, what you are asking is already in my to do list, but it may take a long time to get there, maybe v0.4? because i do not have an another RC522 at the moment and there are other things that really really need to be done before this.

**Comment by akeilox** [comment]
Hi @omersiar understand, I been side tracked now catching up from stored email notifications and code updates, this is a great learning tool with very cool practical applications.

I saw the progress on SPIFF, and was wondering if you will consider adding support for paired-access.
Say like for a store room or file room, where two cards (person) need to be tapped to gain access.
When cardid stored in EPROM, perhaps it can have an optional pair field (name,cardid,pairid... ), and if this field is filled with another card id then mcu will await that card to be scanned too, to give access.
Is it something that can be factored in to the design for future implementation.
I am interested to hear your thoughts on this.

**Comment by omersiar** [comment]
Yeah totaly esp-rfid can work like this, you can divide users into two groups, one have direct access, one have to wait for another person to validate access, actually you can do it now, for second group, you do not even have to deal with eeprom or know them. Just give users rfid tags, and do not let them access, give security guy the valid tag, lets people to enter

**Comment by omersiar** [comment]
i assume there is a security guy, but if there are multiple parties for access, i do not know how to actually maintain this. design your work flow, perfect it, then coding will be easy.

**Comment by rneurink** [comment]
> Are they any plans to change the SS and RST pin selection (esp if rst is redundant now) in GUI to Enter and Exit pin selection so it can support 2 mfrc522 units connected to the same mcu? Perhaps the default is Enter and Exit being optional.
So one can do logs like;

I've got 2 readers so i will work on this after i get the SD card with logging working.

**Comment by omersiar** [comment]
This was done there nicely: https://github.com/torehc/carontepass-v2/blob/master/arduino/caronte-esp8266/caronte-esp8266-two-rfid.ino

**Comment by omersiar** [comment]
@rneurink Also I am still trying to convince my self about timed entry.

I am thinking about global Open-Close hours in Settings Page 

https://codepen.io/jasesmith/pen/reLEyb

and also beside from that setting, every user can also be individually configured as you did it. I need somehow more easy approach for Timed Access, for example selecting boxes for hours of day and day of week like they are doing with torrent clients. 

https://www.raymond.cc/blog/schedule-bittorrent-to-automatically-start-and-stop-downloading/

Also for future reference MomentJS is very promising: https://momentjs.com/

**Comment by omersiar** [comment]
Timed Access (Entry) will likely be in V0.4
